validmind/unit_metrics/composite.py

```diff
@@ -0,0 +1,159 @@
+# Copyright © 2023-2024 ValidMind Inc. All rights reserved.
+# See the LICENSE file in the root of this repository for details.
+# SPDX-License-Identifier: AGPL-3.0 AND ValidMind Commercial
+
+from dataclasses import dataclass
+from typing import Protocol
+from uuid import uuid4
+
+from ..utils import run_async
+from ..vm_models.test.metric import Metric
+from ..vm_models.test.metric_result import MetricResult
+from ..vm_models.test.result_summary import ResultSummary, ResultTable
+from ..vm_models.test.result_wrapper import MetricResultWrapper
+from . import run_metric
+
+
+class MetricProtocol(Protocol):
+    unit_metrics: list[str]
+
+    def output(self, results):
+        pass
+
+
+@dataclass
+class CompositeMetric(Metric):
```

Comment:
I suggest adding a simple description() implementation by default so that the user can take over on the UI and customize it.